flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/FineGrainedSlotManager.java

```diff
@@ -111,6 +111,8 @@
 
     @Nullable private ScheduledFuture<?> lastResourceRequirementsCheck;
 
+    @Nullable private CompletableFuture<Void> requirementsCheckFuture;
```

Comment:
I wonder if we can remove `lastResourceRequirementsCheck`. Currently, it is only used for canceling the check on closing. Alternatively, we can check `started` in `checkResourceRequirements`.